presidio-anonymizer/tests/anonymizers/test_mask.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+
+from presidio_anonymizer.anonymizers import Mask
+from presidio_anonymizer.entities import InvalidParamException
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "text, masking_char, chars_to_mask, from_end, anonymized_text",
+    [
+        ("text", "*", 4, False, "****"),  # Fully mask
+        ("text", "*", 4, True, "****"),  # Fully mask from the end
+        ("text", "*", 3, False, "***t"),  # Partially mask
+        ("text", "*", 3, True, "t***"),  # Partially mask from the end
+        ("text", "*", 5, False, "****"),  # Fully mask with overflowing chars_to_mask
+        ("text", "*", 5, True, "****"),  # from the end
+        ("t", "*", 1, False, "*"),  # Fully mask one character text
+        ("t", "*", 1, True, "*"),  # from the end
+        ("t", "*", 3, False, "*"),  # with overflowing chars_to_mask
+        ("t", "*", 3, True, "*"),  # from the end
+        ("text", "😈", 4, False, "😈😈😈😈"),  # Mask with 'Unicode EmojiSources' character
+        ("text", "*", 0, False, "text"),  # Nullified mask
```

Comment:
Do we have any tests where the text is empty?

Reply:
Should be checked in the engine? when start == end?

Reply:
You're right!

Reply:
but not at the engine.

Reply:
Giving it another thought, empty replacement might be valid again if we don't call from the API. Will add the tests, not implementation change needed